### ISSUE TITLE
Can't resolve argument when using class based API

### ISSUE THREAD

**Issue by f440** [rmosolgo/graphql-ruby#2537]
- graphql-ruby v1.9.13
- Ruby on rails v5 and v6

I made class based type and query:

```
# app/graphql/types/company_type.rb
module Types
  class CompanyType < Types::BaseObject
    field :id, ID, null: false
  end
end

# app/graphql/types/query_type.rb
module Types
  class QueryType < Types::BaseObject
    field :company, CompanyType, null: false do
      argument :id, ID, required: true
    end
    def company(id:)
      Company.find(id)
    end
  end
end

```

and executed query like this:

```
{
  company(id: "1") {
    id
  }
}
```

and caught this error:

```
no implicit conversion of #<Class:0x00007fad86640718> into Hash
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/schema/field.rb:519:in `block in resolve'
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/schema/field.rb:648:in `with_extensions'
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/schema/field.rb:504:in `resolve'
/Users/fuse/go/src/github.com/f440/rails-sandbox/app/graphql/types/base_field.rb:6:in `resolve_field'
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/schema/member/instrumentation.rb:80:in `call'
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/schema/member/instrumentation.rb:80:in `call'
/Users/fuse/go/src/github.com/f440/rails-sandbox/.direnv/ruby/gems/graphql-1.9.13/lib/graphql/field.rb:248:in `resolve'
...
```

I find this in graphql-ruby source:
https://github.com/rmosolgo/graphql-ruby/blob/de26425c8ba2e67c068bbf2c396b95b6847d2ff7/lib/graphql/schema/field.rb#L519

`extended_args` is class (ref. https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/query/arguments.rb ), not hash. `extended_args` can not accept double splat operator.

**Comment by johanandre** [comment]
+1

**Comment by anoopmadhav** [comment]
+1

**Comment by rmosolgo** [comment]
Hi, sorry to hear it! I tried to reproduce the bug with this little script but I didn't succeed: 

```ruby 
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem "graphql", "1.9.13"
end

puts GraphQL::VERSION

class TestSchema < GraphQL::Schema
  class Company < GraphQL::Schema::Object
    field :id, ID, null: false
  end

  class Query < GraphQL::Schema::Object
    field :company, Company, null: true do
      argument :id, ID, required: true
    end

    def company(id:)
      OpenStruct.new(id: id)
    end
  end

  query(Query)
end

pp TestSchema.execute("{ company(id: \"1\") { id } }")
```

I got a successful output: 

```
~/code/graphql-ruby $ ruby test.rb
1.9.13
#<GraphQL::Query::Result @query=... @to_h={"data"=>{"company"=>{"id"=>"1"}}}>
```

Could you try that script locally and see if you can replicate the error? 

Also, I see in your stack trace some customization: 

```
/Users/fuse/go/src/github.com/f440/rails-sandbox/app/graphql/types/base_field.rb:6:in `resolve_field'
```

Could you share the source of that customization?

**Comment by gersmann** [comment]
+1

**Comment by f440** [comment]
Thank you for your response!


> Could you try that script locally and see if you can replicate the error?
> 
> Also, I see in your stack trace some customization:
> 
>    /Users/fuse/go/src/github.com/f440/rails-sandbox/app/graphql/types/base_field.rb:6:in `resolve_field'
> 
> Could you share the source of that customization?


I executed your script on my local machine and got the same response. `Company` is inherit `GraphQL::Schema::Object` in your script, but my one which made by `rails generate graphql:object Company` is inherit `Types::BaseObject`. `Types::BaseObject` is specified `field_class Types::BaseField` and `Types::BaseField` define `resolve_field`.


Source code is here:
- Github: https://github.com/f440/graphql-ruby-2537
- Docker image: https://cloud.docker.com/u/f440/repository/docker/f440/graphql-ruby-2537

```
docker run --rm f440/graphql-ruby-2537 bundle exec rails runner 'GraphqlSampleSchema.execute "{ company(id: \"1\") { id } }"'

**Comment by omarvelous** [comment]
I as well am getting the same issue... Thought I was going crazy...

**Comment by rmosolgo** [comment]
@f440, thanks very much for contributing useful debugging info. I didn't remember that the generator made code like that! I think the problem will be fixed in #2541. Could you try it on master, for example: 

```ruby 
gem "graphql", github: "rmosolgo/graphql-ruby", ref: "master"
```

And let me know how it goes for you? If it works, I'll release 1.9.14 on Monday.

**Comment by f440** [comment]
I confirmed that the problem was fixed on master branch. Thanks a lot!
